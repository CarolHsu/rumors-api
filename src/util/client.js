import config from 'config';
import elasticsearch from 'elasticsearch';

export default new elasticsearch.Client({
  host: config.get('ELASTICSEARCH_URL'),
  log: 'trace',
});
