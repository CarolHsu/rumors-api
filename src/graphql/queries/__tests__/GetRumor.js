import client from 'util/client';
import GraphQL from '../../../../test/util/GraphQL';
import fixtures from '../__fixtures__/GetRumor';

describe('GetRumor', () => {
  let items;
  beforeAll(() => client.bulk({body: fixtures}).then((r) => { items = r.items; }));

  it('should get rumor & associated answers from ID', async () => {
    expect(await GraphQL(`{
      GetRumor(id: "foo") {
        text
        answers {
          versions {
            text
            reference
          }
        }
      }
    }`)).toMatchSnapshot();
  });

  afterAll(() => console.log('rRRrrrRR', items));
});
