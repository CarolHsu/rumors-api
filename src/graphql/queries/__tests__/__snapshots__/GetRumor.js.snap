exports[`GetRumor should get rumor & associated answers from ID 1`] = `
Object {
  "data": Object {
    "GetRumor": Object {
      "answers": Array [
        Object {
          "versions": Array [
            Object {
              "reference": "barbar",
              "text": "bar",
            },
          ],
        },
      ],
      "text": "foofoo",
    },
  },
}
`;
